### PR TITLE
Added Modmap.js as modular file for loading all modules

### PR DESCRIPTION
**Pull request by nkwsy**
Moved content from main.js to modmap related to /modmap.

When testing I commented out main.js to make sure it worked along side. I was not able to get the plantplacement.js to work. 

I also think a commit I did overroad changes charlie made in main.js so worked from this file. https://github.com/charliegroll/Wildmile/blob/d65c7aa4b761cf6d26b9c8c94790e3713caf185a/public/js/main.js

Still need to move all content to plantplacement.js

### CODE REVIEW

File: public/js/modMap.js [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+    console.log(local);
+    var text = draw.text(local.toString()).attr({
+      opacity: op,
+      x: i * 20 - 6,
+      y: n * 60 + 12
+    }).rotate(90);
+  }
+
+};
+// function modMap() {
```
Comment (charliegroll) [inline public/js/modMap.js:179]: do we want to keep this commented out?

Comment (nkwsy) [reply public/js/modMap.js:179]: I actually should just delete it. The previous version used the plants to derive the modules placed which did not really make sense if we had empty modules

**Comment by nkwsy** [comment]
I did not yet remove from main.js, but I did comment it out so it is not being used now. I have added a second commit which 

1. displays the individual module layout in /module/x&y
2. displays the plant in each module
3. changes dropdown to the corresponding plant
4. Modularizes plantPlacement.js
5. Makes the plants have # related to the corresponding drop-down menu (instead of colors)

This does not currently allow for updating, must change the submit logic back to a single form. 
Does not allow for deleting plants.
Allows for users to add a second plant over one which is already planted. 

Kinda tired so may have silly commented out areas and such :)

File: public/js/modMap.js (hunk 1 above)
Comment (charliegroll) [reply public/js/modMap.js:179]: yeah, I agree. if you need this code again, you have it in the commit history. let's avoid checking in commented code if possible

File: public/js/plantPlacement.js [hunk 2]
```diff
@@ -1,86 +1,223 @@
...
-          color,
+
+  function drawMod(shape) {
+    let article = document.querySelector('#circle_1');
+    //  let data = JSON.parse(article.dataset.plantedPlants)
+      console.log(article.dataset.plantedplants);
+    var draw = SVG().addTo('#circle_1').size(900, 300)
+    console.log(shape);
+    if (shape == 'Rtriangle') {
+      var triangle = draw.polygon('0,0 900,1300 900,0').fill('green').stroke({
```
Comment (charliegroll) [inline public/js/plantPlacement.js:58]: why the change from `let` to `var`?

File: public/js/plantPlacement.js [hunk 3]
```diff
@@ -1,86 +1,223 @@
...
+      })
+    }
+    if (shape == 'Ltriangle') {
+      var triangle = draw.polygon('0,0 900,300 900,0').fill('green').stroke({
+        width: 1
+      })
+    }
+    if (shape == 'rectangle') {
+      let rect = draw.rect(900, 300).attr({
+        fill: 'green'
```
Comment (charliegroll) [inline public/js/plantPlacement.js:69]: hex values seem like they'd be preferred - why this change?

File: public/js/plantPlacement.js [hunk 4]
```diff
@@ -1,86 +1,223 @@
...
+    //
+
+    var fillcolor = ['yellow', 'red', 'blue']
+    var fillcolor = ['#7FD674', '#A9E079', '#BAC977', '#E0DB79', '#D6C874'];
+    pselect = [];
+  for (var n = 0; n < 10; n++) {
+    for (var i = 0; i < 4; i++) {
+        // array[i]
+        console.log('ii',n,i);
+        plant = checkPlants(n, i);
```
Comment (charliegroll) [inline public/js/plantPlacement.js:83]: where is `plant` initialized?

File: views/modules.pug [hunk 5]
```diff
@@ -143,6 +143,8 @@ block content
               button.btn.btn-success( type='submit' )
                 i.fas.fa-plus.fa-sm
                 | Submit
+    append partials/scripts 
+      script(src='/js/modMap.js')
```
Comment (charliegroll) [inline views/modules.pug:147]: do we already load `plantPlacement` on the `module` template?

File: public/js/modMap.js [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+  return false;
+}
+
+function modMap() {
+  var draw = SVG().addTo('#modMap').size(400, 3800)
+  var article = document.querySelector('#modMap');
+  console.log(article.dataset);
+  var allMods = JSON.parse(article.dataset.mods);
+  var defaultColor = 'grey'
+  console.log(allMods[0]["x"]);
```
Comment (charliegroll) [inline public/js/modMap.js:28]: there's a chance that this `console.log` statement could break the whole script because `allMods[0]` could be null or undefined - I would recommend either printing out all of `allMods` if you're really interested in debugging it or checking to make sure `allMods[0]` exists first

File: public/js/modMap.js [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+
+  //var rect = draw.rect(60, 20).attr({ fill: '#189968' }).stroke({ color: '#5ECCA2',  width: 2 })
+  var wide = 30;
+  for (var n = 0; n < 30; n++) {
+    for (var i = 0; i < 10; i++) {
+      // array[i]
+      color = '#189968'
```
Comment (charliegroll) [inline public/js/modMap.js:35]: where's this initialized?

File: public/js/modMap.js [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+      var x = 200 - n
+      var y = i
+      //getValueByKey("x", data)
+      // if (article[0].["x"] == x) {
+      //      console.log(x,'found');
+      //  }
+      t = checkIfModExists(allMods, x, i)
```
Comment (charliegroll) [inline public/js/modMap.js:42]: where's this initialized?

File: public/js/modMap.js [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+            }).data('key', {
+              x,
+              y
+            }).stroke({
+              color: '#5ECCA2',
+              width: 1
+            })
+
+          }
+          var rect = draw.rect(20, 60).attr({
```
Comment (charliegroll) [inline public/js/modMap.js:97]: how many `rect` variables are we gonna have here? lol

File: public/js/modMap.js [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+            }).data('key', {
+              x,
+              y
+            }).stroke({
+              color: '#5ECCA2',
+              width: 1
+            })
+
+          }
+          if (t["orientation"] == "Decend") {
```
Comment (charliegroll) [inline public/js/modMap.js:82]: presumably, this should be `else if`

File: public/js/modMap.js [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+      }
+
+      var g = i * 20
+      var h = n * 60
+
+      rect.click(function () {
+        document.getElementById("x").value = this.data('key')["x"];
+        document.getElementById("y").value = this.data('key')["y"];
+        modpage = 'module/'+this.data('key')["x"]+'&'+this.data('key')["y"];
+        window.location.href = modpage;
```
Comment (charliegroll) [inline public/js/modMap.js:150]: once you do this, anything executed next might not happen

File: public/js/plantPlacement.js [hunk 12]
```diff
@@ -1,86 +1,188 @@
...
+let plantMatrix = [];
+let uniquePlants = [];
+let data = {};
+
+let article = document.querySelector('#circle_1');
+
+try {
+  data = JSON.parse(article.dataset.plantedplants);
+}
+catch(e) {
```
Comment (charliegroll) [inline public/js/plantPlacement.js:12]: we should probably handle this better - maybe an error message/dialog?

File: public/js/plantPlacement.js [hunk 13]
```diff
@@ -1,86 +1,188 @@
...
+function updateSelector(selectorName,plantId) {
+  $(selectorName).val(plantId);
+}
+function checkPlants(x, y) {
+  for (var k = 0; k < len; k++) {
+    if (data[k] ) {
+        if (data[k].y == y && data[k].x == x) {
+          val = data[k].plant;
+          console.log(selectors);
+          a = uniquePlants.indexOf(data[k].plant.toString());
```
Comment (charliegroll) [inline public/js/plantPlacement.js:33]: where's this initialized? also, highly recommend naming variables something human-readable and contextual

File: public/js/plantPlacement.js [hunk 14]
```diff
@@ -1,86 +1,188 @@
...
+  $(document).ready(function() {
+    makeSelector(1, 'orange');
+    makeSelector(2, 'red');
+    makeSelector(3, 'yellow');
+    makeSelector(4, 'green');
+    makeSelector(5, 'blue');
+    defineUniquePlants();
+    drawMod('rectangle');
+    console.log(uniquePlants);
+    function individualPlants() {
```
Comment (charliegroll) [inline public/js/plantPlacement.js:183]: where's this fn used? it's a bit misleading to have it inside the document.ready since it's not used here
